From 711022a49c26260db7998b7705b0993c3d81bb21 Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Wed, 21 Dec 2022 08:48:57 +0100
Subject: [PATCH 31/40] Revert "maple_tree: use mt_node_max() instead of direct
 operations mt_max[]"

This reverts commit c5b6a486ec8ea02eb36614db82f2eb8bba3b8d75.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 lib/maple_tree.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/maple_tree.c b/lib/maple_tree.c
index 7f7f2b49f..e297742ec 100644
--- a/lib/maple_tree.c
+++ b/lib/maple_tree.c
@@ -6726,7 +6726,7 @@ static void mt_dump_range64(const struct maple_tree *mt, void *entry,
 
 		if (i < (MAPLE_RANGE64_SLOTS - 1))
 			last = node->pivot[i];
-		else if (!node->slot[i] && max != mt_node_max(entry))
+		else if (!node->slot[i] && max != mt_max[mte_node_type(entry)])
 			break;
 		if (last == 0 && i > 0)
 			break;
@@ -6833,7 +6833,7 @@ void mt_dump(const struct maple_tree *mt)
 	if (!xa_is_node(entry))
 		mt_dump_entry(entry, 0, 0, 0);
 	else if (entry)
-		mt_dump_node(mt, entry, 0, mt_node_max(entry), 0);
+		mt_dump_node(mt, entry, 0, mt_max[mte_node_type(entry)], 0);
 }
 EXPORT_SYMBOL_GPL(mt_dump);
 
-- 
2.39.0.rc2.1.gbd5df96b79

